### PR TITLE
Add total size limit in TxPool

### PR DESCRIPTION
**Pull request by Kourin1996**
# Description

This PR adds TxPool limit by size.
When add new remote tx into tx pool, discard tx with lowest gas price.

Please merge after https://github.com/0xPolygon/polygon-sdk/pull/151

Doc PR: 0xPolygon/polygon-sdk-docs#18

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have tested this code
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

File: command/helper/helper.go [hunk 1]
```diff
@@ -28,6 +28,8 @@ const (
 	DefaultChainID        = 100
 	DefaultPremineBalance = "0x3635C9ADC5DEA00000" // 1000 ETH
 	DefaultConsensus      = "pow"
+	DefaultPriceLimit     = 1
```
Comment (zivkovicmilos) [inline command/helper/helper.go:31]: Why would the default value be `1`?

File: e2e/framework/testserver.go [hunk 2]
```diff
@@ -241,6 +241,18 @@ func (t *TestServer) Start(ctx context.Context) error {
...
+	if len(t.Config.Locals) > 0 {
+		args = append(args, "--locals", strings.Join(t.Config.Locals, ","))
+	}
+
+	if t.Config.NoLocals {
+		args = append(args, "--nolocals")
+	}
+
+	if t.Config.PriceLimit != nil {
+		args = append(args, "--price-limit", strconv.FormatUint(*t.Config.PriceLimit, 10))
```
Comment (zivkovicmilos) [inline e2e/framework/testserver.go]: What do you think about adding support for this value to be passed in as either a decimal or a hex value?

Comment (Kourin1996) [reply e2e/framework/testserver.go]: Actually we should discuss about this in https://github.com/0xPolygon/polygon-sdk/pull/151.
I think it's good idea. I will do there

File: command/helper/helper.go (hunk 1 above)
Comment (Kourin1996) [reply command/helper/helper.go:31]: Actually we should discuss about this in https://github.com/0xPolygon/polygon-sdk/pull/151
In order to block tx with 0 gas price via gossip

Comment (zivkovicmilos) [reply command/helper/helper.go:31]: Got it, thank you for clarifying 👍

**Review by zivkovicmilos — APPROVED** [review]
Thank you for clearing up the questions I've had 🙏 

Good to go 🚀

**Review by lazartravica — CHANGES_REQUESTED** [review]
Excellent work @Kourin1996 

Just 2 more requirements:

- You are lacking test cases for adding local transactions and discarding remote ones 
- We need a corresponding PR in the https://github.com/0xPolygon/polygon-sdk-docs/

**Comment by Kourin1996** [comment]
Doc PR: https://github.com/0xPolygon/polygon-sdk-docs/pull/18

**Comment by Kourin1996** [comment]
> You are lacking test cases for adding local transactions and discarding remote ones

Added at https://github.com/0xPolygon/polygon-sdk/pull/168/commits/bc681d8e091b6bd4d35aad9209f9967a2417d98c

**Comment by lazartravica** [comment]
@Kourin1996 There are merge conflicts that popped up since we are using squash & merge on the repository.

I suggest that next time you open a PR with the target branch set to the one that you have the dependency on, and not develop so that it's easier for you.

Thank you!

**Review by lazartravica — APPROVED** [review]
👌 please resolve the merge conflicts and we are good to go (there are merge conflicts on the docs repo as well).

**Pull request merged**